include/nuttx/progmem.h

```diff
@@ -49,96 +49,127 @@ extern "C"
 
 /****************************************************************************
  * Name: up_progmem_isuniform
  *
  * Description:
- *  Is program memory uniform or erase page and read/write page size differs?
+ *  Are all erasable units of memory the same size?
+ *
+ * Returned Value:
+ *   true if all erasable units are the same size, false if they vary in size
  *
  ****************************************************************************/
 
 bool up_progmem_isuniform(void);
 
 /****************************************************************************
- * Name: up_progmem_pagesize
+ * Name: up_progmem_writegranularity
  *
  * Description:
- *   Return read/write page size
+ *   Return the number of bytes of the smallest allowable write operation.
+ *   All writes MUST be a multiple of the granularity. Some FLASH memories
+ *   require write operations in multiples of 8-bit, 16-bit, 32-bit, 128-bit,
+ *   256-bit, etc. The caller must ensure the size of the data passed into
+ *   up_progmem_write is an even multiple of the writegranularity. Padding
+ *   may be required.
+ *
+ * Returned Value:
+ *   Required granularity of a write operation for the FLASH memory.
  *
  ****************************************************************************/
 
-size_t up_progmem_pagesize(size_t page);
+size_t up_progmem_writegranularity(void);
 
 /****************************************************************************
  * Name: up_progmem_erasesize
  *
  * Description:
- *  Return erase block size. Must be a multiple of the read/write page size.
+ *   Return the erase size for a specified index. Will always be a multiple
+ *   of the write granularity.
+ *
+ * Input Parameters:
+ *   index - ordinal number of the set of erasable units of memory
+ *
+ * Returned Value:
+ *   Erase size for erasable unit of memory, or negative value on error.
+ *
+ *   The following errors are reported (errno is not set!):
+ *
+ *     -EFAULT: On invalid unit of memory
  *
  ****************************************************************************/
 
-size_t up_progmem_erasesize(size_t block);
+ssize_t up_progmem_erasesize(size_t index);
 
 /****************************************************************************
- * Name: up_progmem_getpage
+ * Name: up_progmem_getindex
  *
  * Description:
- *   Address to read/write page conversion
+ *   Get the index of memory, given a FLASH memory address
  *
  * Input Parameters:
  *  addr - Address with or without flash offset
- *         (absolute or aligned to page0)
+ *         (absolute or aligned to the zeroeth unit of memory)
  *
  * Returned Value:
- *   Page or negative value on error.
+ *   Index or negative value on error.
  *   The following errors are reported (errno is not set!):
  *
  *     -EFAULT: On invalid address
  *
  ****************************************************************************/
 
-ssize_t up_progmem_getpage(size_t addr);
+ssize_t up_progmem_getindex(size_t addr);
```

Comment:
In the original naming, the word "block" referred to smallest erasable unit of flash, and "page" referred to the smallest writable unit. This separation in naming should be kept consistent; On this line the function name uses the word "index" referring to original "block" (smallest erasable unit). This should really return the "write granule" and not the eraseblock!

So the function name here would be wrong.

Reply:
No.. this is literally the problem. You interpreted the getpage call as the smallest writable block. But if you look at the majority of the existing drivers, this returned the index of the erasable unit.

Reply:
If this is the intention, the function should be "up_progmem_getblock"